public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -356,14 +377,14 @@ function getVizPanelQueryOptions(vizPanel: VizPanel): QueryOptionsSpec {
   return queryOptions;
 }
 
-function createElements(panels: PanelKind[]): Record<string, PanelKind> {
+function createElements(panels: Element[]): Record<string, Element> {
   return panels.reduce(
     (acc, panel) => {
-      const key = getVizPanelKeyForPanelId(panel.spec.id);
+      const key = panel.kind === 'Panel' ? getVizPanelKeyForPanelId(panel.spec.id) : panel.spec.uid;
```

Comment:
for library panels, maybe we could prefix them similarly to panels, but `library-<uid>` ? The uid itself doesn't say much when reading through the json.

Reply:
I checked this part, and we are having a dependency on the panel menu with this `key` property. Checking the function
 https://github.com/grafana/grafana/blob/79cd8ac894a2917312a256576cf54f540fc5c494/public/app/features/dashboard-scene/utils/utils.ts#L30

 I see we rely on having the key as : `panel-`, this is causing that edit, and inspect do not work for library panels